src/frameworks/opencl/context.rs

```diff
@@ -21,6 +21,55 @@ pub struct Context {
     queue: Option<Queue>
 }
 
+/// The individual properties for `ContextInfo::Properties`
+#[derive(PartialEq, Debug, Clone, Copy)]
+pub enum ContextProperties {
+    /// Identifies a context's associated platform.
+    Platform(Platform),
+    /// Does the user need to sync between frameworks manually (OpenCL 2.0 and later)
+    InteropUserSync(bool),
+}
+
+/// OpenCL context info constants, these ones are for the Query
+#[derive(PartialEq, Debug, Copy, Clone)]
+pub enum ContextInfoQuery {
+    /// Number of references to the context currently held.
+    ReferenceCount,
+    /// Number of devices in the context.
+    NumDevices,
+    /// The properties the context was configured with.
+    Properties,
+    /// The devices (IDs) in the context.
+    Devices
+}
+
+/// OpenCL context info types. Each variant is returned from the same function,
+/// `get_context_info`.
+#[derive(Clone, PartialEq, Debug)]
+pub enum ContextInfo {
+    /// Number of references to the context currently held.
+    ReferenceCount(u32),
+    /// Number of devices in the context.
+    NumDevices(u32),
+    /// The properties the context was configured with.
+    ///
+    /// These are:
+    ///
+    /// - CL_CONTEXT_PLATFORM
+    /// - CL_CONTEXT_D3D10_DEVICE_KHR
+    /// - CL_GL_CONTEXT_KHR (OS-dependent)
+    /// - CL_EGL_CONTEXT_KHR (OpenGL-ES context on embedded devices)
+    /// - CL_GLX_DISPLAY_KHR (GL context on Linux)
+    /// - CL_WGL_HDC_KHR (GL context on Windows)
+    /// - CL_CGL_SHAREGROUP_KHR (GL context sharegroup on MacOS)
```

Comment:
this list is not accurate anymore I think--I was using a reference from 1.1 and the [2.0 docs](https://www.khronos.org/registry/OpenCL/sdk/2.0/docs/man/xhtml/clCreateContext.html) have much more.

for now maybe just mention there are other properties in a link but only the first two are supported right now?